src/main/java/baritone/utils/BaritoneProcessHelper.java

```diff
@@ -33,6 +34,11 @@ public BaritoneProcessHelper(Baritone baritone) {
         baritone.getPathingControlManager().registerProcess(this);
     }
 
+    @Override
+    public void onLostControl(){
+        ExampleBaritoneControl.nextCommand();
```

Comment:
you have a reference to the Baritone here, so it shouldn't be an issue to make examplebaritonecontrol a field

however, i don't think that's reeeeeeeally the way to go, since examplebaritonecontrol is supposed to be an example of how to use the baritone api

tell you what, add a new event to the gameeventlistener like onProcessCompleted(IBaritoneProcess completed), and then this will call `baritone.getGameEventListener.onProcessCompleted(this)`, and examplebaritonecontrol's constructor adds itself as a gameeventlistener and listens for that event,and uses that to advance the commands.

so scratch everything i said above about making examplebaritonecontrol a field of baritone,don't do that, do this instead lol

Reply:
> you have a reference to the Baritone here, so it shouldn't be an issue to make examplebaritonecontrol a field
> 
> however, i don't think that's reeeeeeeally the way to go, since examplebaritonecontrol is supposed to be an example of how to use the baritone api
> 
> tell you what, add a new event to the gameeventlistener like onProcessCompleted(IBaritoneProcess completed), and then this will call `baritone.getGameEventListener.onProcessCompleted(this)`, and examplebaritonecontrol's constructor adds itself as a gameeventlistener and listens for that event,and uses that to advance the commands.
> 
> so scratch everything i said above about making examplebaritonecontrol a field of baritone,don't do that, do this instead lol

👍